packages/pando-lib/test/scripts.spec.ts

```diff
@@ -0,0 +1,190 @@
+        date: '2017-06-12T23:22:12+02:00',
+      },
+      committer: {
+        name: 'John Doe',
+        email: 'johndoe@example.com',
+        date: '2017-06-12T23:22:12+02:00',
+      },
+      message: 'Change nothing\n',
+    }
+
+    const commit2Blob = IPLDGit.util.serialize(commit2Node)
+    commit2Cid = await IPLDGit.util.cid(commit2Blob)
+
+    const nodes = [blobNode, treeNode, commitNode, commit2Node]
+    const result = ipld.putMany(nodes, multicodec.GIT_RAW)
+    ;[blobCid, treeCid, commitCid, commit2Cid] = await result.all()
+  })
+
+  describe('> Commit class', () => {
+    describe('# static get(cid) method - retrieves commit', () => {
+      it('retreives commit node with correct cid', async () => {
+        const cid = await ipld.put(commitNode, multicodec.GIT_RAW)
+
+        commitNode['@cid'] = cid
+        commitNode['@sha'] = cidToSha(cid).toString('hex')
+
+        const node = await Commit.get(cid)
+
+        expect(node).to.deep.equal(commitNode)
+      })
+      it('retreives commit node with parents', async () => {
+        const cid = await ipld.put(commit2Node, multicodec.GIT_RAW)
+
+        commit2Node['@cid'] = cid
+        commit2Node['@sha'] = cidToSha(cid).toString('hex')
+
+        const node = await Commit.get(cid)
+        expect(node).to.deep.equal(commit2Node)
+      })
+    })
+    describe('# put() method - upload to ipfs', () => {
+      it('serializes Commit class and uploads it to ipfs', async () => {
+        const cid = await ipld.put(commitNode, multicodec.GIT_RAW)
+        const node = await Commit.get(cid)
+
+        const newCid = await node.put()
+
+        expect(newCid).to.deep.equal(cid)
+      })
+    })
+    describe('# extend() method - get files from tree', () => {
+      it('recurses through the tree and return paths with cids', async () => {
+        const cid = await ipld.put(commitNode, multicodec.GIT_RAW)
+        const node = await Commit.get(cid)
+
+        await node.extend()
+        expect(node.tree).to.deep.equal({ // Test existing tree node with renamed field
+          'somefile': {
+            blob: blobCid,
+            mode: '100644'
+          }
+        })
+      })
+    })
+  })
+
+  describe('> Tree class', () => {
+    describe('# static get(cid) method - retrieves commit', () => {
+      it('retreives tree node with correct cid', async () => {
+        const cid = await ipld.put(treeNode, multicodec.GIT_RAW)
+        const node = await Tree.get(cid)
+
+        expect(node.entries.somefile).to.deep.equal(treeNode['somefile'])
+      })
+    })
+    describe('# put() method - upload to ipfs', () => {
+      it('serializes Tree class and uploads it to ipfs', async () => {
+        const cid = await ipld.put(treeNode, multicodec.GIT_RAW)
+        const node = await Tree.get(cid)
+
+        const newCid = await node.put()
+
+        expect(newCid).to.deep.equal(cid)
+      })
+    })
+  })
+  
+  describe('> Branch class', () => {
```

Comment:
If we wanna test the `Branch` class and the `fetchOrderedHistory` function properly we should test it we commits which have more than one parent [to check how things are ordered when there are parallel branches]